### DISCUSSION TITLE
Please use github action annotations only for high priority warnings

### DISCUSSION THREAD

**Discussion by tothxa** [Homebrew/discussions#5602] [Everyday usage] [answered]
### Output of `brew config`

```shell
I don't use brew directly, only on github runners
```

### Output of `brew doctor`

```shell
I don't use brew directly, only on github runners
```

### Description of issue

I've noticed recently, that our CI workflow macos builds get warnings from homebrew. (after trying to dive in your PR history, I guess it's mostly the fault of https://github.com/Homebrew/brew/pull/17255)

Most of the warnings are about packages already installed by default (e.g. `git 2.46.0 is already installed and up-to-date. To reinstall 2.46.0, run: brew reinstall git`). At least these can be disabled with `--quiet`. So far so good, though I don't think these are worth the super-escalation of annotations (I mean, e.g. `apt-get` treats already installed packages as a complete non-issue), but it's your software.

Then some of them are recommending "casks" instead of "formulas" (e.g. `Treating doxygen as a formula. For the cask, use homebrew/cask/doxygen or specify the `--cask` flag.`). For whatever reason, you consider this so important that `--quiet` doesn't silence it. But, to put it bluntly, I don't care about your "casks" and "formulas" or whatever, just give it to me from wherever you like. But OK, I tried to replace all these with `homebrew/cask/<name>` as suggested in the message.

Now, that then warns me about disabling Gatekeeper: `macOS's Gatekeeper has been disabled for this Cask`. I guess this one's a pretty big deal for actual macos users, but I can't really do anything about it on a github runner.

And at this point I'm stuck, I see no way to get rid of warning messages that are only polluting our workflow annotations. I guess I could try unsetting `GITHUB_ACTIONS` for the call to `brew` to never have to care about them any more, but I wouldn't mind getting escalated warnings about package deprecations — which seem to have been the original trojan horse for all this spam.

I guess, as far as I'm concerned, hiding the formula vs. cask thing behind `--quiet` or a new stronger extension to it would solve my problem. But my guess is that `homebrew` is a very minor concern for most github action users like me, who just use it to get dependencies installed for their CI runs, so I'd really appreciate if you'd reconsider your usage of github action annotations, and limit it to problems that are really important for this use case. (TBH I'm more inclined to think that actually no program should mess with them at all, so even the deprecation case is pushing it)

**Comment by p-linnane** [comment]
This is one of the most entitled and rude discussion posts I think I've ever seen on here. I'm not sure what is going on in your head that makes you think it's acceptable to act like this. We run this project in our spare time, for free.

Please see this helpful post from our Project Leader entitled [Open Source Maintainers Owe You Nothing](https://mikemcquaid.com/open-source-maintainers-owe-you-nothing/), and really take it to heart. Maybe it will help you understand how to appropriately communicate with OSS maintainers in the future.

**Comment by tothxa** [comment]
I apologise if it sounds more rude than I intended. But I also admit that I am quite a bit frustrated by the amount of time I spent trying to solve this. Given that, I thought it was quite toned down. I apologise, if it was not enough.

The point my style tries to make is that annotations are quite a big cost to anyone trying to keep their workflow runs clean, and that dealing with package managers for dependencies is about the third order of importance for them — not just me. (first is the development of their own program, second are dependencies and CI, and the package manager is only a dependency of the second order problems) Again, I apologise if this still sounds too harsh.

I also apologise for forgetting to actually ask for what I'd like. So could you please reconsider which warnings are worth escalating as workflow annotations and what cost you impose on your users with such annotations?

The project I like to work on as a hobby, and which I'm referring to, is Free Software just as well, so I'm perfectly aware of you not owing me or anyone else anything. Sorry for the misunderstanding.

But I don't think what I ask for is a lot of effort. I'm only asking you to please consider (partially) reverting one commit, that as far as I can see, went in kind of tentatively without much discussion — so I'm trying to reopen the discussion with some user feedback. And I'm sorry if this sounds rude again, I really don't mean it like that.

**Comment by smessmer** [comment]
I don't want to get into a discussion about tone, but I agree with the merit of the request. I just ran into the same issue and think it is not useful that homebrew's warnings hide more important warnings in the Github Action UI. Some way to silence them would be nice, or even better make them opt-in instead of opt-out.

Each of my ~20-30 jobs fired 2-3 of these warnings, all duplicate, and it caused me to miss some actually important warnings where a job wasn't performing as it should.

**Comment by carlocab** [comment, marked as answer]
Homebrew/brew#18329 should silence most of these annotations for you.

> Then some of them are recommending "casks" instead of "formulas" (e.g. `Treating doxygen as a formula. For the cask, use homebrew/cask/doxygen or specify the `--cask` flag.`). For whatever reason, you consider this so important that `--quiet` doesn't silence it. But, to put it bluntly, I don't care about your "casks" and "formulas" or whatever, just give it to me from wherever you like. But OK, I tried to replace all these with `homebrew/cask/<name>` as suggested in the message.

You can silence these by using `brew install --formula` or `brew install --cask`. If you don't care which one: I suggest using `brew install --formula`. Though I tend to agree that `--quiet` should probably silence these warnings.

**Reply by carlocab** [reply]
Warnings of the form `Treating doxygen as a formula.` will be silenced by `--quiet` if Homebrew/brew#18330 is merged.

**Reply by tothxa** [reply]
> [Homebrew/brew#18329](https://github.com/Homebrew/brew/pull/18329) should silence most of these annotations for you.

Thank you! :) I see it's already merged too, that was real quick! :) I hope it soon makes it to the deployed runner images too.

> You can silence these by using `brew install --formula`

Oops, indeed, I forgot about that option. Thank you for the tip! (the warning suggests that casks are preferable, so it looks like I concentrated too much on that direction)

> Warnings of the form `Treating doxygen as a formula.` will be silenced by `--quiet` if https://github.com/Homebrew/brew/pull/18330 is merged.

Thank you again! With that, all parts of my original post are addressed.

**Reply by MikeMcQuaid** [reply]
> [Homebrew/brew#18329](https://github.com/Homebrew/brew/pull/18329) should silence most of these annotations for you.

Note: this is being reverted, sorry: https://github.com/Homebrew/brew/pull/18329

I agree with you, though, @tothxa that there's a lot of noisy warnings that we should address and we will aim to do so.

**Reply by tothxa** [reply]
Thank you for the notice. Since it had been pointed out that there's a possible workaround that I missed, there's no immediate urgency.
